src/main/java/earl/util/Storage.java

```diff
@@ -0,0 +1,77 @@
+package earl.util;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import earl.exceptions.EarlException;
+import earl.tasks.Task;
+
+/**
+ * Class responsible for reading and writing data to disk.
+ */
+public class Storage {
+
+    private final String filePath;
+    private final ArrayList<Task> tasks = new ArrayList<>();
+
+    /**
+     * Class constructor.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Returns an {@code ArrayList} of {@code Task} read from the disk.
+     * <p>
+     * Attempts to find the storage file at the given file path.
+     * Starts with an empty file if no existing file is found.
+     *
+     * @return an {@code ArrayList} of {@code Task} read
+     * @throws EarlException if the file could not be created or read
+     */
```

Comment:
I like that your JavaDoc comments are well elaborated for the more complex methods!